src/Life/Github.hs

```diff
@@ -20,44 +19,49 @@ module Life.Github
        , removeFromRepo
        , updateDotfilesRepo
        , updateFromRepo
+       , branchState
 
          -- * Constants
        , master
+       , setCurrentBranch
        ) where
 
 import Control.Exception (catch, throwIO)
 import Path (Abs, Dir, File, Path, Rel, toFilePath, (</>))
 import Path.IO (copyDirRecur, copyFile, getHomeDir, withCurrentDir)
 import System.IO.Error (IOError, isDoesNotExistError)
 
-import Life.Configuration (LifeConfiguration (..), lifeConfigMinus, parseRepoLife)
+import Lens.Micro.Platform ((^.))
+import Life.Configuration (Branch(..), BranchState(..), LifeConfiguration (..), branch, lifeConfigMinus, parseRepoLife, parseHomeLife)
 import Life.Message (chooseYesNo, errorMessage, infoMessage, warningMessage)
 import Life.Shell (lifePath, relativeToHome, repoName, ($|))
 
-newtype Owner  = Owner  { unOwner  :: Text } deriving (Show)
-newtype Repo   = Repo   { unRepo   :: Text } deriving (Show)
-newtype Branch = Branch { unBranch :: Text } deriving (Show)
+import qualified Data.Text as T
+
+newtype Owner = Owner { unOwner  :: Text } deriving (Show)
+newtype Repo  = Repo  { unRepo   :: Text } deriving (Show)
 
 ----------------------------------------------------------------------------
 -- VSC commands
 ----------------------------------------------------------------------------
 
-askToPushka :: Text -> IO ()
-askToPushka commitMsg = do
+askToPushka :: Branch -> Text -> IO ()
+askToPushka branch'@(Branch branchname) commitMsg = do
```

Comment:
+ to camelCase, just did not notice.
About ' - it is hlint recommendation, because `branch` is lens for `LifeConfiguration`.

Reply:
> because branch is lens for 

Ah, shit. Yeah, we don't want to have name shadowing. But still looks like `branch` variable name is used more often then the correspdonding field in the configuration. I propose to rename field in `LifeConfiguration`

Reply:
I'll try to implement your requirements below, and maybe avoid ' here.